src/elastio-snap.c

```diff
@@ -4557,17 +4651,18 @@ static int __tracer_setup_cow(struct snap_device *dev, struct block_device *bdev
 			if(!fallocated_space){
 				max_file_size = size * SECTOR_SIZE * elastio_snap_cow_fallocate_percentage_default;
 				do_div(max_file_size, 100);
-				dev->sd_falloc_size = max_file_size;
-				do_div(dev->sd_falloc_size, (1024 * 1024));
 			}else{
 				max_file_size = fallocated_space * (1024 * 1024);
-				dev->sd_falloc_size = fallocated_space;
 			}
 
 			//create and open the cow manager
 			LOG_DEBUG("creating cow manager");
 			ret = cow_init(dev, cow_path_full, SECTOR_TO_BLOCK(size), COW_SECTION_SIZE, dev->sd_cache_size, max_file_size, uuid, seqid, &dev->sd_cow);
 			if(ret) goto error;
+
+			dev->sd_falloc_size = dev->sd_cow->file_max;
+			do_div(dev->sd_falloc_size, (1024 * 1024));
```

Comment:
We update `dev->sd_cow->file_max` in `cow_init`, so I've put this code here to update `dev->sd_falloc_size` appropriately